drivers/volume/portworx/schedops/k8s-schedops.go

```diff
@@ -684,14 +684,23 @@ func (k *k8sSchedOps) IsPXReadyOnNode(n node.Node) bool {
 	}
 	// Need to make sure if px pod is present or not
 	for _, pod := range pxPods.Items {
-		if pod.Labels["name"] == PXDaemonSet {
-			isPxPodPresent = true
-		}
-
 		if pod.Labels["name"] == PXDaemonSet && !k8sCore.IsPodReady(pod) {
 			printStatus(k, pod)
 			return false
 		}
+
+		if pod.Labels["name"] == PXDaemonSet {
+			isPxPodPresent = true
+			// Update pod restart count
+			containersStatuses := pod.Status.ContainerStatuses
+			for _, containerStatus := range containersStatuses {
```

Comment:
Here initially n.PxPodRestartCount = 0
After reboot test pod has restart count =1
    => n.PxPodRestartCount will become 1 (0+1)
2. After second reboot pod has restart count =2
3. node n.PxPodRestartCount will become 3 rt? (1+2)
is my understanding wrong?

Reply:
Fixed. It should be updated with pod restart count instead of adding up the restart count.

Reply:
if we are taking pod restart count ,how are we differentiating between actual restart and error restarts?